cmd/notary/tuf.go

```diff
@@ -546,3 +587,30 @@ func getRemoteTrustServer(config *viper.Viper) string {
 	}
 	return defaultServerURL
 }
+
+func getTrustPinning(config *viper.Viper) (trustpinning.TrustPinConfig, error) {
+	var ok bool
+	// Need to parse out Certs section from config
+	certMap := config.GetStringMap("trust_pinning.certs")
+	resultCertMap := make(map[string][]string)
+	for gun, certSlice := range certMap {
+		var castedCertSlice []interface{}
+		if castedCertSlice, ok = certSlice.([]interface{}); !ok {
+			return trustpinning.TrustPinConfig{}, fmt.Errorf("invalid format for trust_pinning.certs")
```

Comment:
I haven't heard a good argument either way. Pointers are more memory efficient but copies (non-pointers) are easier to reason about. As with json.RawMessage there are cases where it really matters due to Golang's addressability concepts. The go docs seem to be unopinionated.

IMO, in this context, the parsing function would be fine returning a pointer, but the caller should go on to pass the config by value to any other receivers (so the receivers can't modify the config in a way other functions will be affected by).